src/test/java/edu/hm/hafner/util/SetTest.java

```diff
@@ -0,0 +1,149 @@
+package edu.hm.hafner.util;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Assertions;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+
+public class SetTest {
+    HashSet<Integer> s;
+
+    @BeforeEach
+    public void setUp() {
+        s = new HashSet<Integer>();
+    }
+
+    @Test
+    void testAdd() {
+        Assertions.assertEquals(true, s.add(Integer.valueOf(9)));
+        Assertions.assertEquals(1, s.size());
+        Assertions.assertNotEquals(true, s.add(Integer.valueOf(9)));
```

Comment:
Für diese Aufgabe war angedacht das JUnit Framework AssertJ zu nutzen. Das bietet einige Vorteile. Hier wäre damit z.B. diese Alternative möglich, die besser zu lesen ist.
```suggestion
        Assertions.assertThat(s.add(9)).isFalse();
```

`Integer.valueOf` muss man hier wegen dem [Java-Mechanismus Autoboxing](https://docs.oracle.com/javase/tutorial/java/data/autoboxing.html) so oder so nicht explizit nutzen.